crates/re_types/src/archetype.rs

```diff
@@ -291,3 +230,61 @@ impl<A: Archetype> crate::LoggableBatch for GenericIndicatorComponent<A> {
 }
 
 impl<A: Archetype> crate::ComponentBatch for GenericIndicatorComponent<A> {}
+
+// ---
+
+/// An arbitrary named [indicator component].
+///
+/// [indicator component]: [`Archetype::Indicator`]
+#[derive(Debug, Clone, Copy)]
+pub struct NamedIndicatorComponent(pub ComponentName);
+
+impl NamedIndicatorComponent {
+    #[inline]
+    pub fn as_batch(&self) -> MaybeOwnedComponentBatch<'_> {
+        MaybeOwnedComponentBatch::Ref(self)
+    }
+
+    #[inline]
+    pub fn to_batch(self) -> MaybeOwnedComponentBatch<'static> {
+        MaybeOwnedComponentBatch::Owned(Box::new(self))
+    }
+}
+
+impl crate::LoggableBatch for NamedIndicatorComponent {
+    type Name = ComponentName;
+
+    #[inline]
+    fn name(&self) -> Self::Name {
+        self.0
+    }
+
+    #[inline]
+    fn num_instances(&self) -> usize {
+        1
+    }
+
+    #[inline]
+    fn arrow_field(&self) -> arrow2::datatypes::Field {
+        let name = self.name().to_string();
+        arrow2::datatypes::Field::new(
+            name.clone(),
+            arrow2::datatypes::DataType::Extension(
+                name,
+                Box::new(arrow2::datatypes::DataType::Null),
+                None,
+            ),
+            false,
+        )
+    }
+
+    #[inline]
+    fn try_to_arrow(&self) -> SerializationResult<Box<dyn arrow2::array::Array>> {
```

Comment:
we could call this just `to_arrow` - we don't usually prefix fallible functions with `try_`